hs/src/Reach/Debugger.hs

```diff
@@ -0,0 +1,100 @@
+{-# OPTIONS_GHC -Wno-missing-export-lists #-}
+
+module Reach.Debugger where
+
+import Control.Monad.Reader
+
+import qualified Data.Map.Strict as M
+import Reach.AST.Base
+import Reach.AST.DLBase
+import Reach.AST.LL
+
+-- NOTE: let several base types be Integer/String as a placeholder for now
+
+type ConsensusEnv = M.Map DLVar DLVal
+
+type Balances = [Double]
+
+type Ledger = M.Map Account Balances
+
+type Frontend = ( Account, SLPart, M.Map DLVar DLVal, [ DLTail ] )
+
+type Frontends = [ Frontend ]
+
+type NewPartActions = M.Map SLPart [ DLTail ]
+
+type State = (Ledger, ConsensusEnv, Frontends, NewPartActions, Action)
+
+-- possible backend actions
+data Action
+  = ConsAction LLConsensus
+  | StepAction LLStep
+
+data Contract = Contract
+  { contr_contract_info :: Integer
+  , address :: String
+  , contr_token :: String
+  }
+
+data Account = Account
+  { nw_acc :: Integer
+  , backend :: LLProg
+  , contract :: Contract
+  , acc_contract_info :: Integer
+  , acc_token :: String
+  }
+
+data DLVal
+  = V_Null
+  | V_Bool Bool
+  | V_UInt Integer
+  | V_Bytes Integer
+  | V_Digest String
```

Comment:
The digest is abstract and wouldn't be a concrete thing. Maybe you'd just expose what it was a digest of (i.e. a DLVal)